configuration/configuration.go

```diff
@@ -89,36 +90,114 @@ const (
 	varLogJSON                              = "log.json"
 	varWITDomainPrefix                      = "wit.domain.prefix"
 	varWITURL                               = "wit.url"
+
+	varServiceAccountConfigName    = "service-account-secrets.conf"
+	varDefaultServiceAccountConfig = "/etc/fabric8/" + varServiceAccountConfigName
 )
 
+type ServiceAccountConfig struct {
+	Accounts []ServiceAccount
+}
+
+type ServiceAccount struct {
+	Name    string   `mapstructure:"name"`
+	Id      string   `mapstructure:"id"`
+	Secrets []string `mapstructure:"secrets"`
+}
+
 // ConfigurationData encapsulates the Viper configuration object which stores the configuration data in-memory.
 type ConfigurationData struct {
+	// Main Configuration
 	v *viper.Viper
+
+	// Service Account Configuration
+	sa *viper.Viper
 }
 
 // NewConfigurationData creates a configuration reader object using a configurable configuration file path
-func NewConfigurationData(configFilePath string) (*ConfigurationData, error) {
+func NewConfigurationData(mainConfigFile string, serviceAccountConfigFile string) (*ConfigurationData, error) {
 	c := ConfigurationData{
-		v: viper.New(),
+		v:  viper.New(),
+		sa: viper.New(),
 	}
+
+	// Set up the main configuration
 	c.v.SetEnvPrefix("AUTH")
 	c.v.AutomaticEnv()
 	c.v.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
 	c.v.SetTypeByDefaultValue(true)
 	c.setConfigDefaults()
 
-	if configFilePath != "" {
+	// Set up the service account configuration (stored in a separate config file)
+	c.sa.SetTypeByDefaultValue(true)
+
+	if mainConfigFile != "" {
 		c.v.SetConfigType("yaml")
-		c.v.SetConfigFile(configFilePath)
+		c.v.SetConfigFile(mainConfigFile)
 		err := c.v.ReadInConfig() // Find and read the config file
 		if err != nil {           // Handle errors reading the config file
 			return nil, errors.Errorf("Fatal error config file: %s \n", err)
 		}
 	}
+
+	// If a service account configuration file has been specified, check that it exists
+	if serviceAccountConfigFile != "" {
+		if _, err := os.Stat(serviceAccountConfigFile); os.IsNotExist(err) {
```

Comment:
This code will fail if the file does exist because in this case err == nil
Should be something like this instead:
```
if serviceAccountConfigFile != "" {
    if _, err := os.Stat(serviceAccountConfigFile); err != nil {
        return nil, err // Return error for any reason if you can't get stat. If it doesn't exist or if it failed for another reason. 
    }
}
```